transports/tls/src/certificate.rs

```diff
@@ -99,26 +95,22 @@ pub fn generate(
     // Endpoints MAY generate a new key and certificate
     // for every connection attempt, or they MAY reuse the same key
     // and certificate for multiple connections.
-    let certificate_keypair = rcgen::KeyPair::generate(P2P_SIGNATURE_ALGORITHM)?;
+    let certificate_keypair = rcgen::KeyPair::generate()?;
```

Comment:
```suggestion
    let certificate_keypair = rcgen::KeyPair::generate_for(P2P_SIGNATURE_ALGORITHM)?;
```
Be explicit about the used alogrithm.

Reply:
yeah makes sense, thanks Elena!